docs/usage-guide/topics/ch04-software-architecture.md

```diff
@@ -0,0 +1,21 @@
+# Software Architecture
+
+User interaction with s2n-tls happens primarily through the `s2n_connection` and `s2n_config` structures.
+
+## Primary Structures
+
+Users start by building a config. In the general case, there will be one config per server. This involves loading the certificate, configuring session resumption, etc. Users should configure an `s2n_config` before associating any `s2n_connection` objects with it.
+
+Users must then configure a connection, and associate it with a config. `s2n_connection` is responsible for managing the actual state of a TLS connection. In a TLS server, there will be one `s2n_connection` for each TCP stream. For each `s2n_config`, there may be many `s2n_connection` structs associated with it.
```

Comment:
Most of this seems like it should go in the config section? I think we describe a lot of this in that section already though.

Reply:
I think the first part should go into the config chapter, and the second one should go into the connection chapter. Since configuration should happen before connection, I would also move the config chapter on top of the connection chapter to make it looks more intuitive.